BLSQ/iaso
# IA-3258 add payment status filter

Explain what problem this PR is resolving

Related JIRA tickets : IA-3258

## Self proofreading checklist


## Doc
JsDoc comments in the code

## Changes
- Backend
    - Add filter in change requests Filterset
    - Move `parse_comma_separated_numeric_values` to common.py and remove duplicate versions
    - Move `PaymentStatuses` outside of `Payment` class for reuse
    - Add paths to LOCALE_PATHS for backend translations
    - Add `locale` folder with translations in `iaso` app folder
    - Add `DropDownOptionsViewSet` and `DropDownOptionsSerializer` for future endpointst that return `TextChoices`
    - Add `PaymentOptionsViewSet`, which extends `DropDownOptionsViewSet`


- Frontend
    - Add translations
    - Add typing for OPTIONS response
    - Add param to change requests route
    - Add `useGetPaymentStatusOptions` (but with GET request) and replace duplicate
    - Add `mapOptions` method in utils to convert OPTIONS response to dropdown list format (+ unit test)
    - Cleanup some old comments
  
## How to test

make sure to run `django-admin compilemessages` when not launching the backend from Docker

- Go change requests page
- Test payments filter
- Change language and check that dropdown is correctly translated

Extra mile:
- Go to polio budget and try to filter by country using multiple values (= testing that `parse_comma_separated_numeric_values` still works as intended)
- Go to Payments, open dialog and try to edit payment status (= test that `useGetPaymentStatusOptions` works as intended)

## Print screen / video

https://github.com/user-attachments/assets/d1dee3b3-e98e-4f8a-8a75-17bcff311368

## Notes

Using the OPTIONS request was causing permission issues, so I made a dedicated endpoint with the `TextChoices`.
I left the `mapOptions` helper in case we figure out how to fix the permissions problem. If not it can be removed.

- Cypress tests are passing: https://github.com/BLSQ/iaso/actions/runs/10214039194
